lib/index.js

```diff
@@ -1,26 +1,57 @@
+'use strict'
+
 const vm = require('vm')
 const merge = require('lodash.merge')
 const cloneDeep = require('lodash.clonedeep')
-const parseAndReplace = require('./expression_parser')
-
-let delimiters, unescapeDelimiters, delimiterRegex, unescapeDelimiterRegex, conditionals, loops, scopes, options
+const placeholders = require('./placeholders')
 
-module.exports = function PostHTMLExpressions (_options = {}) {
-  options = _options
-  // set up delimiter options and detection
-  delimiters = options.delimiters || ['{{', '}}']
-  unescapeDelimiters = options.unescapeDelimiters || ['{{{', '}}}']
-  delimiterRegex = new RegExp(`.*${delimiters[0]}(.*)${delimiters[1]}.*`, 'g')
-  unescapeDelimiterRegex = new RegExp(`.*${unescapeDelimiters[0]}(.*)${unescapeDelimiters[1]}.*`, 'g')
+let delimitersSettings = []
+let conditionals, loops, scopes, options
 
-  // identification for delimiter options, for the parser
-  delimiters.push('escaped')
-  unescapeDelimiters.push('unescaped')
+function escapeRegexpString (input) {
+  return input.replace(/[|\\{}()[\]^$+*?.]/g, '\\$&')
+}
 
-  // conditional and loop options
-  conditionals = options.conditionalTags || ['if', 'elseif', 'else']
-  scopes = options.scopeTags || ['scope']
-  loops = options.loopTags || ['each']
+module.exports = function PostHTMLExpressions (_options) {
+  // set default options
+  options = Object.assign({
+    delimiters: ['{{', '}}'],
+    unescapeDelimiters: ['{{{', '}}}'],
+    conditionalTags: ['if', 'elseif', 'else'],
```

Comment:
Are options for setting the conditional-, loop-, scope tags really necessary? what if someone tries to add e.g options.loopTags = ['each', 'foreach', 'forof', 'forin'...] expecting them to work like the normally do in JS ? :) To reduce eventual confusion and configuration imo, we should maybe consider removing them from options. cc @voischev @posthtml/collaborators